core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -337,6 +338,7 @@ private boolean canContainDeletedFiles(ManifestFile manifest) {
     return canContainExpressionDeletes || canContainDroppedPartitions || canContainDroppedFiles || canContainDropBySeq;
   }
 
+  @SuppressWarnings("CollectionUndefinedEquality")
```

Comment:
Are the suppressions still needed now that a different set class is used?

Reply:
unfortunately yes. ErrorProne still thinks the issue around `CollectionUndefinedEquality`, but those are false positives